### DISCUSSION TITLE
Implementation of the RSS feed

### DISCUSSION THREAD

**Discussion by SamuelNitsche** [christophrumpel/larastreamers#127] [Ideas]
I just took a look at the rss feed and noticed that this does not really make sense for this kind of application.

Rss feeds are generally used for items published in the past. The streams are **upcoming** items.

The current implementation uses the `updated_at` field of each stream as the publish date which displays upcoming streams as published at a date in the past.

If we use the real `scheduled_start_time`, most rss readers will use the current time or only display the items once the `scheduled_start_time` is in the past.

This is how the current stream is displayed in Reeder 5.
![IMG_0033](https://user-images.githubusercontent.com/24483576/118814781-cbed1800-b8b0-11eb-9817-8808941ca653.jpeg)

**Comment by christophrumpel** [comment]
Thanks, @SamuelNitsche. I don't use RSS feeders. I just pushed a commit to provide tests for the current solution. (so we have a base to refactor) and I removed the dummy summary with the description.

Would you use the RSS feed for such an application?

**Comment by SamuelNitsche** [comment]
In general I would use the feed but I can't think of a way to get it working well.

**Comment by christophrumpel** [comment]
@freekmurze any ideas what could be a good solution here?

**Comment by freekmurze** [comment]
@SamuelNitsche couldn't we use the stream creation date as the publish date of a feed item? This way you would see it in your RSS reader as soon as the stream is created.

Feel free to submit a PR for this 👍

**Comment by SamuelNitsche** [comment]
@freekmurze That's what I already tried. The problem of this approach is that most rss readers display the current date instead of the date in the future. So you see a stream that starts tomorrow but the feed says it starts **now**.

**Comment by christophrumpel** [comment]
@freekmurze any idea of a better solution? I will implement an archive for this app soon, maybe it makes more sense to create the RSS feed for only past streams?

**Comment by freekmurze** [comment]
I'm thinking it is just a matter of clear copy. The creation date of the RSS item shouldn't necessarily be the start of the stream. In the body of the feed item you could mention the real start date of the stream.

**Comment by SamuelNitsche** [comment]
Is there a real benefit of the rss feed then? I think the main goal of larastreamers is to have a place to see all streams (in the future).

**Comment by freekmurze** [comment]
Yeah, personally I'd love to see announcements in my RSS reader so that I don't have to visit the site. But ultimately @christophrumpel decides if it's worth it for him 👍

**Comment by christophrumpel** [comment]
If it benefits you @freekmurze, we should keep it for sure. Since the app is about upcoming streams, we should try to make it work for those also in the RSS feed.

So how can we improve the current implementation? Anyone up for a PR for that?

**Comment by GertjanRoke** [comment]
What about a Telegram channel instead? Something like Laravel News has.

**Comment by christophrumpel** [comment]
For me, this is very hard to tell since I don't use an RSS reader.

Please @freekmurze decide how it works best.

@GertjanRoke Telegram could be interesting too, but probably not yet.

**Comment by prikkprikkprikk** [comment]
I recently posted a couple of issues about the feed, but was asked to contribute here instead.

I use the RSS feed, but I don't use it to _keep track of when_ streams will happen. I mainly want to know who's streaming and what it's about. Like @SamuelNitsche says, future dates doesn't really make sense for RSS feeds.

My suggestion for the date is to just include the stream date/time in the entry description. (If one wanted to go fancy, maybe there even could be a link to add it to your calendar …)

**Comment by prikkprikkprikk** [comment]
There's another issue in the RSS feed, though: It looks like the author's name instead displays the title:

<img width="580" alt="Screenshot 2021-12-06 at 17 17 49" src="https://user-images.githubusercontent.com/147902/144882240-18a974cc-bf64-44dc-a27f-4ac7156c8f49.png">

My guess is a copy/paste that wasn't changed afterwards …

**Reply by christophrumpel** [reply]
@prikkprikkprikk in the last PR we fixed the issue about the author.

Regarding the overall idea of what the stream should be for, this is still not decided. Some want to only know about streams that already happened, some prefer to hear about only upcoming streams. So it is not decided yet. (just an update for you)

**Comment by JonPurvis** [comment]
I think it might be difficult to have 1 feed that works with how people use RSS feeds, so could we not have multiple feeds and let people subscribe to the one that is best suited for them?

I'm in the same boat as @christophrumpel though as I also don't use RSS feeds.
